### PR TITLE
New feature Jest tests

### PR DESCRIPTION
**Pull request by miguelesco**
## What's new 

- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components.
- All the tests pass.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @miguelesco 👋

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable linebreak-style */
 import react from 'react';
 import {
-  BrowserRouter as Router,
+  HashRouter as Router,
```
Comment (mikenath223) [inline src/App.jsx:4]: - [ ] There is a failing test; `App.test.js`, kindly update that test case as it doesn't relate to this project.

File: src/components/calculator/calculator.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import React from 'react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import Calculator from './calculator';
+
+describe('Calculator', () => {
+  it('should render without crashing', () => {
+    const { getByText } = render(<Calculator />);
+    expect(getByText('Lets do some Math')).toBeInTheDocument();
+  });
```
Comment (mikenath223) [inline src/components/calculator/calculator.test.js:17]: - [x] Please add tests based on user interactions using the React Testing Library.

Comment (mikenath223) [reply src/components/calculator/calculator.test.js:17]: - [x] Kindly ensure it stimulates different user interactions covering as many operations as possible.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "styled-components": "^5.3.1",
     "web-vitals": "^1.1.2"
   },
-  "homepage": "https://miguelesco.github.io/",
+  "homepage": "https://miguelesco.github.io/math-magicians",
```
Comment (mikenath223) [inline package.json:20]: - [x] In the readme, kindly include instructions to run tests.

File: src/App.jsx (hunk 1 above)
Comment (henatan99) [reply src/App.jsx:4]: - This issue is resolved. Well done! :clap:

File: src/components/calculator/calculator.test.js (hunk 2 above)
Comment (henatan99) [reply src/components/calculator/calculator.test.js:17]: - This issue is well addressed. Thank you for making the requested changes. :+1: :clap:

File: package.json (hunk 3 above)
Comment (henatan99) [reply package.json:20]: - This issue is well addressed. Nice job! :+1: :100:

**Review by henatan99 — APPROVED** [review]
Hi @miguelesco , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**